scheduler/src/cook/prometheus_metrics.clj

```diff
@@ -0,0 +1,134 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+
+;; Declares prometheus metrics for cook scheduler.
+
+(ns cook.prometheus-metrics
+  (:require [iapetos.core :as prometheus]
+            [iapetos.export :as prometheus-export]
+            [mount.core :as mount]))
+
+;; Quantiles are specified as a map of quantile to error margin.
+(def default-summary-quantiles {0.25 0.01 0.5 0.01 0.75 0.01 0.9 0.005 0.95 0.005 0.98 0.001 0.99 0.001 0.999 0.0001})
+
+;; We define all the metric names here to get IDE support and avoid the chance of runtime
+;; errors due to misspelled metric names.
+;; We are standardizing the metric format to be :cook/<component>-<metric-name>-<unit>
+(def scheduler-rank-cycle-duration :cook/scheduler-rank-cycle-duration-seconds)
```

Comment:
I'm on the fence on whether we simplify this by using a macro or not.

(defmacro defprometheus
    [name]
    `(def ~name ~(keyword "cook" (str name)))
    
If we can teach the IDE to recognize these (via that macro feature I showed you) as definitions for searching and such then I'd vote for defining this macro and cleaning up the duplication here.